### DISCUSSION TITLE
Create Presentations/PowerPoints using Marp

### DISCUSSION THREAD

**Discussion by JMcrafter26** [open-webui/open-webui#7853] [Ideas]
I came across [Marp](https://github.com/marp-team/marp) (a tool to create presentation slides with Markdown) and thought, it would be really handy to be able to generate presentations directly in OpenWebUI.

Example use case:

> You have a chat about the Ancient Greek and you want a way to visualize this information. You could click the make presentation button, and you would get a visually appealing presentation as a PDF/HTML/PowerPoint.

Implementation:
- You would need to have Marp CLI installed, as a NPM package or binary.
- Get the chat history and generate a Marp presentation markdown using an AI agent
- Then pass the markdown to Marp CLI and let it create the slides and pass them back to the user

**Comment by ips972** [comment]
It would be great if this feature were integrated as a built-in option within the Open WebUI, eliminating the need for any external dependencies or additional installations. I've set up Open WebUI for my team, utilizing the local Ollama instance, and the most common question I receive is: **"Can I use it to create PowerPoint presentations?"** It's amusing how many people dread making presentations! Nonetheless, **adding this functionality would certainly be a popular addition.**

**Reply by JMcrafter26** [reply]
I made a very basic script to generate presentations using Ollama and Marp:
https://github.com/JMcrafter26/Marp-AI
